syncer/syncer.go

```diff
@@ -1617,7 +1612,17 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 
 	sql := strings.TrimSpace(string(ev.Query))
 	usedSchema := string(ev.Schema)
-	parseResult, err := s.parseDDLSQL(sql, ec.parser2, usedSchema)
+	parser2 := parser.New()
```

Comment:
maybe could encapsulate into a function 🤔

Reply:
NOTE: most of `QueryEvent` events are not a DDL but maybe something like `BEGIN`, so how about only get this parser only if it's a DDL to avoid unnecessarily parse for `StatusVars` (in other words, but get and use parser into `parseDDLSQL`).

Reply:
@lance6716 do not miss this comment. ⬆️

Reply:
seems we should parse it first to decide if it's a DDL. Parsing for `StatusVars` just read it once, not sure that affect performance much